packages/xo-web/src/xo-app/sr/tab-general.js

```diff
@@ -32,9 +166,104 @@ export default class TabGeneral extends Component {
     }
   }
 
-  render() {
-    const { sr, vdis, vdiSnapshots, unmanagedVdis } = this.props
+  _getDiskGroups = createSelector(
+    () => this.props.vdis,
+    () => this.props.vdiSnapshots,
+    () => this.props.unmanagedVdis,
+    (vdis, vdiSnapshots, unmanagedVdis) => {
+      const groups = []
+      const parentDisks = []
+      const snapshotsByVdi = groupBy(vdiSnapshots, '$snapshot_of')
+      const unmanagedVdisById = keyBy(unmanagedVdis, 'id')
+      let baseCopiesUsage = 0
+      let bc
+      let nBaseCopies = 0
+      let root
+      const vdisByRoot = groupBy(
+        vdis.map(({ parent, ...vdi }) => {
+          baseCopiesUsage = 0
+          nBaseCopies = 0
+          root = undefined
+          if (parent !== undefined) {
+            bc = unmanagedVdisById[parent]
+            while (bc !== undefined) {
+              nBaseCopies++
+              baseCopiesUsage += bc.usage
+              root = bc.id
+              parentDisks.push(bc)
+              bc = unmanagedVdisById[bc.parent]
+            }
+          }
+          return {
+            ...vdi,
+            baseCopiesUsage,
+            nBaseCopies,
+            root,
+          }
+        }),
+        'root'
+      )
+
+      const baseCopiesNotAttached = differenceBy(
+        unmanagedVdis,
+        parentDisks,
+        'id'
+      )
+      if (baseCopiesNotAttached.length > 0) {
+        groups.push({
+          ...baseCopiesNotAttached[0],
+          usage: sumBy(baseCopiesNotAttached, 'usage'),
+        })
+      }
 
+      let snapshotsNotAttached
+      if ((snapshotsNotAttached = snapshotsByVdi[undefined]) !== undefined) {
+        snapshotsNotAttached.forEach(snapshot => {
+          groups.push(snapshot)
+        })
+      }
+
+      let allSnapshots
+      let snapshots
+      let usage
+      forEach(vdisByRoot, (vdis, key) => {
+        if (key === 'undefined') {
```

Comment:
We should not have such a case IMO. The root should be the VDI itself in that case.